cmd/dcos/main.go

```diff
@@ -26,13 +29,45 @@ func main() {
 		Fs:         afero.NewOsFs(),
 	})
 
-	if len(os.Args) == 2 && os.Args[1] == "--version" {
+	if err := run(ctx, os.Args); err != nil {
+		os.Exit(1)
+	}
+}
+
+// run launches the DC/OS CLI with a given context and args.
+func run(ctx api.Context, args []string) error {
+	var verbosity int
+	var showVersion bool
+
+	// Register the version and verbose global flags.
+	globalFlags := pflag.NewFlagSet(args[0], pflag.ContinueOnError)
+	globalFlags.SetOutput(ioutil.Discard)
```

Comment:
Why do we discard errors here?

Reply:
This hybrid solution with a flagset outside of cobra is not the most robust. I've added some comments about it. Hopefully we'll clean this out when moving away from cobra at some point.